application/models/Permission.php

```diff
@@ -720,15 +724,35 @@ private static function comparePermissionTitle($aApermission, $aBpermission)
     public function getUserId($iUserID = null)
     {
         if (empty($iUserID)) {
-            if (Yii::app() instanceof CConsoleApplication) {
-                /* Alt : return 1st forcedAdmin ? */
+            if (App() instanceof CConsoleApplication) {
+                /* Alt : return 1st forcedSuperAdmin ? */
                 throw new Exception('Permission must not be tested with console application.');
             }
-            /* See TestBaseClass tearDownAfterClass */
-            $iUserID = Yii::app()->session['loginID'];
+            return self::getCurrentUserId();
         }
         return $iUserID;
     }
+
+    /**
+     * get the current id of connected user
+     * @return int|null user id
+     * @throws Exception
+     */
+    public static function getCurrentUserId()
+    {
+        if (App() instanceof CConsoleApplication) {
+            return null;
```

Comment:
I wonder how this would impact Email Sending crons... isn't there any perission checking there?

Reply:
No, you don't have to check Permission in console